lighthouse-core/config/config.js

```diff
@@ -314,19 +314,18 @@ class Config {
   }
 
  /**
-  * Filter out any unrequested items from the config, based on requested top-level aggregations.
-  * @param {!Object} oldConfig Lighthouse config object
-  * @param {!Array<string>} aggregationIDs Id values of aggregations to include
-  * @return {Object} new config
-  */
-  static generateNewConfigOfAggregations(oldConfig, aggregationIDs) {
+   * @param {!Object} config
+   * @param {!Array<string>} categoryIds
+   * @return {!Object}
+   */
+  static generateNewConfigOfCategories(config, categoryIds) {
     // 0. Clone config to avoid mutating it
-    const config = JSON.parse(JSON.stringify(oldConfig));
-    // 1. Filter to just the chosen aggregations
-    config.aggregations = config.aggregations.filter(agg => aggregationIDs.includes(agg.id));
+    config = Object.assign({}, config);
```

Comment:
I dislike the idea of random parts of config not supporting a consistent feature set a little more than being careful about mutation, but I'll revert for now